### ISSUE TITLE
Merging of config files in umbrella project not working as expected

### ISSUE THREAD

**Issue by coneybeare** [rrrene/credo#803]
### Precheck

* Proposals for new features should be submitted via: https://github.com/rrrene/credo-proposals
* For bugs, please do a quick search and make sure the bug has not yet been reported here

### Environment

* Credo version (`mix credo -v`): 1.4.0
* Erlang/Elixir version (`elixir -v`): Elixir 1.10.4 (compiled with Erlang/OTP 23)
* Operating system: MacOS 10.15.5

### What were you trying to do?

Merge 2 config files in an umbrella project, allowing customization of rules on a specific app.

### Expected outcome

I would expect the files to be merged

### Actual outcome

They are not merged.

---

Using a structure of:

````
- umbrella_app
  - .credo.exs
  - apps
    - sub_app
      - .credo.exs
````

A root level `umbrella_app/.credo.exs`:

````elixir
%{
  configs: [
    %{
      name: "default",
      checks: [
        # Not compatible with our version of elixir
        {Credo.Check.Refactor.MapInto, false},
        {Credo.Check.Warning.LazyLogging, false},
        # Team choice to allow TODO comments
        {Credo.Check.Design.TagTODO, false}
      ]
    }
  ]
}
````

And a `umbrella_app/apps/sub_app/.credo.exs`:

````elixir
%{
  configs: [
    %{
      name: "sub_app",
      checks: [
        # We disable credo violations in this auto-generated code
        {Credo.Check.Readability.ModuleDoc, false}
      ]
    }
  ]
}
````

Shouldn't this disable the ModuleDoc check in the sub_app? Instead, I get:

````bash
$ mix credo
Checking 34 source files ...

  Code Readability
┃
┃ [R] → Use `@moduledoc false` if a module will not be documented.
┃       apps/sub_app/lib/sub_app/model/source_response.ex:5:11 #(SubApp.Model.SourceResponse)
┃ [R] → Use `@moduledoc false` if a module will not be documented.
┃       apps/sub_app/lib/sub_app/model/source_event_response.ex:5:11 #(SubApp.Model.SourceEventResponse)
┃ [R] → Use `@moduledoc false` if a module will not be documented.
┃       apps/sub_app/lib/sub_app/model/source_event_attributes.ex:5:11 #(SubApp.Model.SourceEventAttributes)
┃ [R] → Use `@moduledoc false` if a module will not be documented.
┃       apps/sub_app/lib/sub_app/model/source_event.ex:5:11 #(SubApp.Model.SourceEvent)
┃ [R] → Use `@moduledoc false` if a module will not be documented.
┃       apps/sub_app/lib/sub_app/model/source_attributes.ex:5:11 #(SubApp.Model.SourceAttributes)
┃  ...  (7 more, use `-a` to show them)

Please report incorrect results: https://github.com/rrrene/credo/issues

Analysis took 0.3 seconds (0.02s to load, 0.2s running 43 checks on 34 files)
91 mods/funs, found 12 code readability issues.
````

**Comment by coneybeare** [comment]
If this is user-error in the config files, perhaps an example of an umbrella app config setup would help me, and future implementors as well.

**Comment by rrrene** [comment]
@coneybeare The `name` field of the config has to be `default` if you want the configs to be merged when running

```
mix credo
``` 

The `name` field provides support for having differently named configs, so you could have a `default` config for development, but have a different config for named `ci` which you can run on your CI server using

```
mix credo --config-name ci
``` 

If you have any input on how this can be made any more clear in the docs, I am all for it :+1:

**Comment by coneybeare** [comment]
Perhaps there is some other issue in my credo setup, because changing the name of the sub app's config to `default` still does not have the intended result.

`umbrella_app/apps/sub_app/.credo.exs`
````elixir
%{
  configs: [
    %{
      # Name must be "default" if we want to merge with upstream configs
      name: "default",
      checks: [
        # We disable credo violations in this auto-generated code
        {Credo.Check.Readability.ModuleDoc, false}
      ]
    }
  ]
}
````

````
$ mix credo
Checking 34 source files ...

  Code Readability
┃
┃ [R] → Use `@moduledoc false` if a module will not be documented.
┃       apps/catalog_api/lib/catalog_api/model/source_response.ex:5:11 #(CatalogAPI.Model.SourceResponse)
┃ [R] → Use `@moduledoc false` if a module will not be documented.
┃       apps/catalog_api/lib/catalog_api/model/source_event_response.ex:5:11 #(CatalogAPI.Model.SourceEventResponse)
┃ [R] → Use `@moduledoc false` if a module will not be documented.
┃       apps/catalog_api/lib/catalog_api/model/source_event_attributes.ex:5:11 #(CatalogAPI.Model.SourceEventAttributes)
┃ [R] → Use `@moduledoc false` if a module will not be documented.
┃       apps/catalog_api/lib/catalog_api/model/source_event.ex:5:11 #(CatalogAPI.Model.SourceEvent)
┃ [R] → Use `@moduledoc false` if a module will not be documented.
┃       apps/catalog_api/lib/catalog_api/model/source_attributes.ex:5:11 #(CatalogAPI.Model.SourceAttributes)
┃  ...  (7 more, use `-a` to show them)

Please report incorrect results: https://github.com/rrrene/credo/issues

Analysis took 0.2 seconds (0.02s to load, 0.2s running 43 checks on 34 files)
91 mods/funs, found 12 code readability issues.

Showing priority issues: ↑ ↗ →  (use `--strict` to show all issues, `--help` for options).
````

**Comment by coneybeare** [comment]
Additional datapoint: running mix credo from the sub app works as intended.

````
 $ mix credo
Checking 17 source files ...

Please report incorrect results: https://github.com/rrrene/credo/issues

Analysis took 0.2 seconds (0.01s to load, 0.2s running 42 checks on 17 files)
56 mods/funs, found no issues.

Use `--strict` to show all issues, `--help` for options.
````

However running from umbrella root does not, the problem I am looking to solve

**Comment by coneybeare** [comment]
@rrrene Is running `mix credo` from the root of the umbrella app supposed to be respecting the credo options of a containing sub app in `apps`?

**Comment by rrrene** [comment]
@coneybeare Missed this question: No, it is not. But I can see how one might think that is how it works. Maybe we should address this in the documentation/terminal output.

**Comment by gseddon** [comment]
I was looking for the same behaviour. I have found that putting this alias into the root `mix.exs` gives you mostly what you want
```
  defp aliases do
    [
      # run `mix credo` in all child apps
      credo: ["cmd mix credo || true"]
    ]
  end
```
The only issue is that the colouring doesn't work properly this way

**Comment by rrrene** [comment]
I think this is now much clearer in the docs, but if you have any more input, just re-open this issue.

Thx again for reporting this! :+1:
